homeassistant/components/hardware/models.py

```diff
@@ -3,10 +3,20 @@
 from __future__ import annotations
 
 from dataclasses import dataclass
-from typing import Protocol
+from typing import TYPE_CHECKING, Protocol, TypedDict
 
 from homeassistant.core import HomeAssistant, callback
 
+if TYPE_CHECKING:
+    from .websocket_api import SystemStatus
+
+
+class HardwareData(TypedDict, total=False):
+    """Hardware data."""
```

Comment:
Note: I went for `TypedDict` instead of `dataclass` to avoid breaking change if other integrations try to access the data

If we are sure it shouldn't be accessed outside the integration, then I can migrate to dataclass

Reply:
Items in `hass.data` should never be accessed directly by other integrations than the integration that put them in `hass.data`.

Reply:
Yes... but I got caught with https://github.com/home-assistant/core/pull/136617

Reply:
I agree. Let's adjust this 👍

Reply:
> I agree. Let's adjust this 👍

Done